app/services/user_importer.rb

```diff
@@ -0,0 +1,16 @@
+require "csv"
+
+class UserImporter
+  def call(csv_path)
+    @csv_path = csv_path
+    upsert_users
+  end
+
+  private def upsert_users
+    CSV.foreach(@csv_path, headers: true) do |row|
+      User.find_or_create_by(email: row[:email]) do |user|
+        user.assign_attributes(row)
```

Comment:
This is my fault for hinting at `find_or_create_by`:

This should be able to be run against an existing user and update their attributes, because `email` has a unique constraint on it this `find_or_create_by` and `assign_attributes` does not appear to work here, we might have to do this manually as we cannot use `upsert` with the SQL adaptor?

As pseudo code:

```
for each row in the csv:
   find the User by email
   if one is found
     update the attributes but not email
   else
     create and save a new User
  end
end
```

Reply:
I actually had a spec covering this, but it turns out the spec wasn't very good 🤦 Thanks for running this and checking!

We can still use `find_or_create_by` but not with a block, we need to do it like this 🤷 

```ruby
user = User.find_or_create_by(email: row[:email])
user.assign_attributes(row)
user.save
```

Fixed this so it should all work as expected now!